### ISSUE TITLE
Stryker JavaScript Mutator- always logs trace

### ISSUE THREAD

**Issue by jmpresso** [stryker-mutator/stryker#592]
### Summary

I recently activated the JavaScript Mutator. It always gives trace logging. Ex:

[2018-01-23 12:02:03.440] [TRACE] JavaScriptMutator - Generated mutant for mutator IfStatement 

### Stryker config

I set the logLevel like I see set in several example conf files here, but it doesn't appear to change anything.

```js
config.set({
        testRunner: 'karma',
        testFramework: 'jasmine',
        reporter: ['html', 'clear-text', 'event-recorder'],
        mutator: 'javascript',
        files: [
            {
                pattern: 'src/**/*.js',
                included: true,
                mutated: true
            },
            'test/**/!(*e2e)spec.js',
            {
                pattern: 'test/**/*.json',
                included: false,
                mutated: false
            }
        ],
        logLevel: 'info',
        thresholds: { high: 99, low: 94, break: 90 }
    });
```

### Stryker environment

```
    "grunt-sonar-runner": "2.4.3",
    "stryker": "^0.17.0",
    "stryker-api": "^0.12.0",
    "stryker-html-reporter": "^0.11.4",
    "stryker-jasmine": "^0.7.3",
    "stryker-javascript-mutator": "^0.3.1",
    "stryker-karma-runner": "^0.11.3",
```

### Your Environment
| software         | version(s)
| ---------------- | -------  
| node             |  v8.x
| npm              |  v5.x
| Operating System | Windows 7

**Comment by simondel** [comment]
Thanks for opening the issue! You're first to open an issue since we added the template!

Is that the exact message that you get? Not: `Generated mutant for mutator ${mutatorName} in file ${file.name} with replacement code "${replacement}"`? (With the template strings replaced of course)

**Comment by nicojs** [comment]
I think i kbow whats going on. The stryker-javascript-mutator creates a logger using `require('log4js').getLogger(....)`. The problem is that, when node modules cannot be flattened, it gets its own (unconfigured) copy of 'log4js'. 

We can fix this by adding a `setGlobalLogLevel` in the javascript mutator. We do this in other plugins, but forgot to add it here.

@simondel, maybe we can think of a more robust way of creating loggers and migrate to the latest version as well. In the mean time, lets add the setGlobalLogLevel call.

**Comment by simondel** [comment]
@nicojs Upgrading to log4js 2x is a great idea that we've been putting off for too long. I misread the summary of this issue, I thought that *only* the IfStatement-Mutator was being logged.

**Comment by simondel** [comment]
This has been released in stryker-javascript-mutator@0.3.3 :) Thanks for reporting it!
